install.js

```diff
@@ -4,37 +4,58 @@ const exec = util.promisify(require("child_process").exec);
 
 const challenges = JSON.parse(fs.readFileSync("challenges.json").toString());
 
+const CREATE_ETH_STABLE_VERSION_FOR_CHALLENGES = "0.1.0";
+const createEthVersion =
+  process.argv[2] || CREATE_ETH_STABLE_VERSION_FOR_CHALLENGES;
+
 const setupChallenge = async (challenge) => {
   try {
-    if (!fs.existsSync(`./${challenge.name}`)) {
-      console.log("====] CLONING " + challenge.name + "[==============]");
-      const result1 = await exec(
-        "git clone -b " +
-          challenge.name +
-          " " +
-          challenge.github +
-          " " +
-          challenge.name
-      );
-      console.log(`stdout: ${result1.stdout}\n`);
-      console.log(`stderr: ${result1.stderr}\n`);
+    const tempFolder = `./${challenge.name}_temp`;
+
+    if (fs.existsSync(tempFolder)) {
+      console.log(`🗑  Removing old temporary folder of ${challenge.name}`);
+      fs.rmSync(tempFolder, { recursive: true, force: true });
     }
 
-    console.log("====] UPDATING " + challenge.name + "[==============]");
-    const result2 = await exec(
-      "cd " + challenge.name + " && git pull && yarn install"
+    console.log(`📦 Installing ${challenge.name} to temporary folder`);
+    const result1 = await exec(
+      `npx --yes create-eth@${createEthVersion} -s hardhat -e scaffold-eth/se-2-challenges:${challenge.name} ${tempFolder}`
     );
-    console.log(`stdout: ${result2.stdout}\n`);
-    console.log(`stderr: ${result2.stderr}\n`);
+    // console.log(`stdout: ${result1.stdout}\n`);
+
+    if (result1.stderr) {
+      console.log(`stderr: ${result1.stderr}\n`);
+    }
+
+    // If installation was successful, swap the folders
+    if (fs.existsSync(tempFolder)) {
+      if (fs.existsSync(`./${challenge.name}`)) {
+        console.log(`🗑  Removing old ${challenge.name} folder`);
+        fs.rmSync(`./${challenge.name}`, { recursive: true, force: true });
+      }
+
+      console.log(`📂 Moving new installation to final location`);
+      fs.renameSync(tempFolder, `./${challenge.name}`);
+    }
   } catch (e) {
-    console.log(e);
+    console.log(`❌ Error installing ${challenge.name}:`, e);
+    // Clean up temp folder if it exists
+    if (fs.existsSync(`./${challenge.name}_new`)) {
```

Comment:
Umm we are not creating any folder named `..._new` right? I think this should have been `..._temp` ?

Reply:
fixed, thanks